### PR TITLE
Send emails about outdated requests

### PR DESCRIPTION
**Pull request by Baggerone**

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -22,9 +22,16 @@ package:
...
 functions:
-  cron:
+  maintenance:
     handler: bin/maintenance
     events:
     # cron(Minutes Hours Day-of-month Month Day-of-week Year)
     # Either `day-of-month` or `day-of-week` must be a question mark (?)
     - schedule: cron(0 4 ? * SUN *)
+
+  outdatedRequests:
```
Comment (briskt) [inline serverless.yml:32]: I don't understand why this needs to be a separate lambda since it's also running once a week.

File: domain/domain.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"time"
+)
+
+func RunTask(url, task string, timeout time.Duration) error {
```
Comment (briskt) [inline domain/domain.go]: It's usually helpful to have refactors, especially ones that are just code moved from one place to another, be done as a separate PR. Or, at least as a separate commit so it's easier to see what changes were made.

File: cron/outdatedrequests/main.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	log.SetFlags(0)
+
+	now := time.Now().UTC()
+	log.Println("WeCarry API Email Notification for Outdated Requests started at", now.Format(time.RFC1123Z))
+
+	url := os.Getenv("SERVICE_INTEGRATION_URL") + "/service"
+	task := "outdated_requests"
+
+	log.Println("running task: " + task)
+	if err := domain.RunTask(url, task, ApiTimeout); err != nil {
```
Comment (briskt) [inline cron/outdatedrequests/main.go]: Why is the timeout passed as a parameter? It looks like both instances are using the same value.

File: domain/domain_test.go [hunk 4]
```diff
@@ -26,7 +27,7 @@ func Test_runTask(t *testing.T) {
 			}))
 			defer server.Close()
 
-			err := runTask(server.URL, tt.task)
+			err := RunTask(server.URL, tt.task, 2*time.Second)
```
Comment (briskt) [inline domain/domain_test.go]: Oh, is this why the timeout is a parameter? Why is it shorter for the test?

**Review by briskt — COMMENTED** [review]
I have a few questions.

File: serverless.yml (hunk 1 above)
Comment (Baggerone) [reply serverless.yml:32]: Having it separate gives us freedom to run it on a different schedule than the other jobs. Although it's set for a weekly run here, maybe in the future we will want to change that.

File: domain/domain_test.go (hunk 4 above)
Comment (Baggerone) [reply domain/domain_test.go]: I figured that it didn't need to be very long for the test and we would rather only wait 2 seconds for the test to fail than 10 seconds.

Comment (Baggerone) [reply domain/domain_test.go]: Maybe that point is irrelevant for tests. In any case, does it really matter that it's different in the test?

Comment (briskt) [reply domain/domain_test.go]: Not sure what would cause a test to fail in a way that would hit the timeout. No problem either way, just seemed unusual.

**Review by briskt — APPROVED** [review]

**Review by briskt — APPROVED** [review]
